typing/env.ml

```diff
@@ -1615,6 +1638,17 @@ let module_declaration_address env id presence md =
   | Mp_present ->
       Lazy_backtrack.create_forced (Aident id)
 
+let module_declaration_address_lazy env id presence md =
```

Comment:
It doesn't look like you actually need `module_declaration_address` anymore, you can just use this one instead.